packages/components/src/mobile/bottom-sheet/header/README.md

```diff
@@ -0,0 +1,67 @@
+# BottomSheet Header
+
+BottomSheet Header components provide styled elements for composing header UI within a `BottomSheet`.
+
+## Usage
+
+```jsx
+import { BottomSheet } from '@wordpress/components';
+
+export default = () => (
+	<BottomSheet>
+		<BottomSheet.Header>
+			<BottomSheet.Header.BackButton onPress={ () => {} } />
+			<BottomSheet.Header.Title>A Sheet Title</BottomSheet.Header.Title>
+			<BottomSheet.Header.ApplyButton onPress={ () => {} } />
+		</BottomSheet.Header>
+	</BottomSheet>
+);
+```
+
+## BottomSheet.Header
+
+Provides structural styles for left-center-right layout for header UI.
+
+## BottomSheet.Header.Title
+
+Displays a styled title for a bottom sheet.
+
+## BottomSheet.Header.ApplyButton
+
+Displays a styled button to apply settings of bottom sheet controls.
+
+### Props
+
+#### onPress
+
+Callback invoked once the button is pressed.
+
+## BottomSheet.Header.BackButton
+
+Displays a styled button to navigate backwards from a bottom sheet.
+
+### Props
+
+#### onPress
+
+Callback invoked once the button is pressed.
+
+## BottomSheet.Header.CancelButton
+
+Displays a styled button to dismiss a full screen bottom sheet.
+
+### Props
+
+#### onPress
+
+Callback invoked once the button is pressed.
+
+## BottomSheet.Header.CloseButton
+
+Displays a styled button to dismiss a full screen bottom sheet.
+
+### Props
+
+#### onPress
+
+Callback invoked once the button is pressed.
```

Comment:
> Apologies if my previous changes removed too much of that, and thus creating more work to add the intent back.

No worries! I think it's great to discuss this and I'm happy this is being addressed. The usage of these patterns will certainly grow, so it's valuable to consider many paths. There is a tendency (and maybe a goal?) to make changes to block settings immediately reflected in the block rendering (i.e. no "confirm" or "cancel" needed). I think this can't cover the case for any fullscreen sheets, though, so this may be _at least one_ source of tension. Historically, though, this may have been addressed in a somewhat ad-hoc way, giving rise to some redundancy in naming — but that's just my guess.

In any case, removing truly synonymous components and clarifying the usage of the remaining ones is a great idea. Also, I especially like [the idea mentioned here](https://github.com/WordPress/gutenberg/pull/34309#discussion_r697477701) to add `iosText` as a prop. This accurate signalling provides immediate clarity, whereas the current form feels like it is violating the principle of least surprise, imo.